File: src/main/java/org/dbshell/commands/connections/dto/ConnectionInfo.java
package org.dbshell.commands.connections.dto;

import java.sql.Connection;

public class ConnectionInfo {
    private String _contextName = "";
    private String _jndiName = "";
    private Connection _connection;

    public ConnectionInfo(String contextName, String jndiName, Connection connection) {
       _contextName = contextName;
       _jndiName = jndiName;
       _connection = connection;
    }

    public String getContextName() {
        return _contextName;
    }

    public String getJndiName() {
        return _jndiName;
    }

    public Connection getConnection() {
        return _connection;
    }
}

File: src/main/kotlin/org/dbshell/commands/connections/dto/ConnectionInfo.kt
package org.dbshell.commands.connections.dto

import org.bradfordmiller.simplejndiutils.JNDIUtils
import org.dbshell.environment.EnvironmentVars
import java.sql.Connection
import java.sql.SQLException

data class ConnectionInfo(val contextName: String, val jndiName: String, val connection: Connection)

class ConnectionInfoUtil {
    companion object {
        @Throws(SQLException::class)
        fun getConnectionFromCurrentContextJndi(): ConnectionInfo {
            val ctxJndi = EnvironmentVars.currentContextAndJndi
            val ctx = ctxJndi.context
            val jndi = ctxJndi.jndi
            val conn = JNDIUtils.getJndiConnection(jndi, ctx)
            return ConnectionInfo(ctxJndi.context, ctxJndi.jndi, conn)
        }
    }
}